agent/src/util/discovery_operator.rs

```diff
@@ -705,34 +627,20 @@ pub mod start_discovery {
         kube_interface: Arc<dyn k8s::KubeInterface>,
         endpoint: &'a DiscoveryHandlerEndpoint,
         dh_details: &'a DiscoveryDetails,
-    ) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+    ) -> anyhow::Result<()> {
         loop {
-            let deregistered;
-            match discovery_operator.get_stream(endpoint).await {
-                Some(stream_type) => {
-                    // Since connection was established, be sure that the Discovery Handler is marked as having a client
-                    discovery_operator.set_discovery_handler_connectivity_status(
-                        endpoint,
-                        DiscoveryHandlerStatus::Active,
-                    );
-                    match stream_type {
-                        StreamType::External(mut stream) => {
-                            match discovery_operator
-                                .internal_do_discover(
-                                    kube_interface.clone(),
-                                    dh_details,
-                                    &mut stream,
-                                )
-                                .await
-                            {
-                                Ok(_) => {
-                                    discovery_operator.set_discovery_handler_connectivity_status(
-                                        endpoint,
-                                        DiscoveryHandlerStatus::Waiting,
-                                    );
-                                    break;
-                                }
-                                Err(status) => {
+            if let Some(stream_type) = discovery_operator.get_stream(endpoint).await {
+                match stream_type {
+                    StreamType::External(mut stream) => {
+                        match discovery_operator
+                            .internal_do_discover(kube_interface.clone(), dh_details, &mut stream)
+                            .await
+                        {
+                            Ok(_) => {
+                                break;
+                            }
+                            Err(e) => {
+                                if let Some(status) = e.downcast_ref::<tonic::Status>() {
                                     if status.message().contains("broken pipe") {
```

Comment:
I like how we simplified the discovery and different sub-status ..
I see we are no longer using SHARED_INSTANCE_OFFLINE_GRACE_PERIOD_SECS before marking the instance as offline?
Are there implications? How will that work for edge scenarios with intermittent connection issues?

Reply:
I noticed you resolved this since there is a loop in the Discovery Operator that periodically runs `delete_offline_instances` but i could image that there may have been confusion around DiscoveryHandlerStatus and InstanceStatus. Another reason to remove the DiscoveryHandlerStatus

Reply:
Thanks Kate, I resolved it as I figured it out myself when I reviewed the rest of the discovery_operator. 
I added a comment with resolution, not sure where it went.